aiida_mlip/calculations/base.py

```diff
@@ -0,0 +1,250 @@
+            help="Mlip architecture to use for calculation, defaults to mace",
+        )
+        spec.input(
+            "model",
+            valid_type=ModelData,
+            required=False,
+            help="Mlip model used for calculation",
+        )
+        spec.input(
+            "struct",
+            valid_type=StructureData,
+            required=False,
+            help="The input structure.",
+        )
+        spec.input(
+            "precision",
+            valid_type=Str,
+            default=lambda: Str("float64"),
+            help="Precision level for calculation",
+        )
+        spec.input(
+            "device",
+            valid_type=Str,
+            required=False,
+            default=lambda: Str("cpu"),
+            help="Device on which to run calculation (cpu, cuda or mps)",
+        )
+
+        spec.input(
+            "log_filename",
+            valid_type=Str,
+            required=False,
+            default=lambda: Str(cls._LOG_FILE),
+            help="Name of the log output file",
+        )
+        spec.input(
+            "metadata.options.output_filename",
+            valid_type=str,
+            default=cls._DEFAULT_OUTPUT_FILE,
+        )
+        spec.input(
+            "metadata.options.input_filename",
+            valid_type=str,
+            default=cls._DEFAULT_INPUT_FILE,
+        )
+        spec.input(
+            "metadata.options.scheduler_stdout",
+            valid_type=str,
+            default="_scheduler-stdout.txt",
+            help="Filename to which the content of stdout of the scheduler is written.",
+        )
+
+        spec.inputs.validator = cls.validate_inputs
+
+        spec.output("std_output", valid_type=SinglefileData)
+        spec.output("log_output", valid_type=SinglefileData)
+        # Exit codes
+        spec.exit_code(
+            305,
+            "ERROR_MISSING_OUTPUT_FILES",
+            message="Some output files missing or cannot be read",
+        )
+        spec.input(
+            "config",
+            valid_type=JanusConfigfile,
+            required=False,
+            help="Name of the log output file",
+        )
+
+    @classmethod
+    def validate_inputs(
+        cls, inputs: dict, port_namespace: aiida.engine.processes.ports.PortNamespace
+    ):
+        """
+        Check if the inputs are valid.
+
+        Parameters
+        ----------
+        inputs : dict
+            The inputs dictionary.
+
+        port_namespace : `aiida.engine.processes.ports.PortNamespace`
+            An instance of aiida's `PortNameSpace`.
+
+        Raises
+        ------
+        ValueError
+            Error message if validation fails, None otherwise.
+        """
+        # Wrapping processes may choose to exclude certain input ports
+        # If the ports have been excluded, skip the validation.
+        if "struct" not in port_namespace and "config" not in port_namespace:
+            raise ValueError("'Structure' namespaces is required.")
```

Comment:
this needs fixing in general caus I should also check inside the config if a structure is given. but I assume janus also raises an error if you don't give a structure, so maybe I can just update the error with 
"at least one between struct or config should be given"

Reply:
Janus would throw an error, but you'd possibly hit other errors first setting up the nodes?

But that's a good point, just checking for `struct` or `config` could still mean you're missing the structure, if it's not in the config. Since it's quite fundamental, you probably want to handle it and not rely on it filtering through and being handled nicely by janus.

I'm not completely clear how the namespaces fit in, but something along the lines of "structure must be specified through struct or config" (with appropriate formatting/language for the namespaces) is probably what you want here, then another check in the config that it contains the structure if you don't have it already?